.github/workflows/continous-integration.yml

```diff
@@ -319,29 +319,57 @@ jobs:
         DD_ARGS: --ddtrace
       run: |
         make ${{ matrix.test }}
+        poetry run coverage xml -o ${{ github.workspace }}/tests/${{ matrix.test }}-coverage.xml --include="rasa/*"
 
-    - name: Send Coverage Report 📊
-      if: needs.changes.outputs.backend == 'true' && matrix.python-version == 3.7 && matrix.os != 'windows-latest'
-      continue-on-error: true
-      env:
-        COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        COVERALLS_SERVICE_NAME: github
-        COVERALLS_PARALLEL: true
-      run: poetry run coveralls
+    - name: Store coverage reports
+      if: needs.changes.outputs.backend == 'true' && matrix.os == 'ubuntu-latest'
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.test }}-coverage.xml
+        path: |
+          ${{ github.workspace }}/tests/${{ matrix.test }}-coverage.xml
 
-  complete_coverage:
-    name: Complete coverage
+
+  upload_coverage_reports:
+    name: Upload coverage reports to codeclimate
     runs-on: ubuntu-20.04
-    needs: [ test, changes ]
-    if: needs.changes.outputs.backend == 'true'
+    needs:
+      - test
+      - changes
 
     steps:
-      - name: Tell coveralls we're finished reporting
-        uses: coverallsapp/github-action@master
-        continue-on-error: true
+      - name: Checkout git repository 🕝
+        if: needs.changes.outputs.backend == 'true'
+        uses: actions/checkout@v2
+
+      - name: Get backend coverage reports
+        if: needs.changes.outputs.backend == 'true'
+        uses: actions/download-artifact@v2
+        with:
+          path: ${{ github.workspace }}/tests_coverage
+
+      - name: Move all xml files to parent dir
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          dir1="${{ github.workspace }}/tests_coverage"
+          subs=`ls ${{ github.workspace }}/tests_coverage`
+          dir2="${{ github.workspace }}/tests_coverage/final"
```

Comment:
A very very minor and not something you necessarily need to agree with: I would prefer something like `sourceDir` and `destinationDir` instead of `dir1` and `dir2` :)

Reply:
ah you're totally right, forgot to clean out the code on here, good catch!